### ISSUE TITLE
NULL and default values problem

### ISSUE THREAD

**Issue by lance-edmodo** [embulk/embulk#430]
Here's the scenario:

Input table is in a Redshift instance. Output table is in a MySQL instance.
Field "id" in the input table has NULL values in some rows.
In the output table, field "id" is declared "NOT NULL DEFAULT '0'".

When Embulk does a table copy, the MySQL instance fails with (paraphrased) "field 'id' cannot be NULL". When Sqoop does a table copy, it succeeds and the "id" field has 0 in those rows where it was NULL in the Redshift instance.

Embulk is:
1) using an INSERT statement that does not trigger the NULL->0 directive and thus attempts to save a NULL to the "id" field.

In other words, Sqoop is either:
1) figuring out the NULL -> 0 directive and doing it on the fly, or
2) using an INSERT statement that happens to trigger the NULL->0 directive correctly.

I can see how to fix this in my Embulk templates, but I have to make a custom template for each table with the problem. Is it possible to fix this problem once inside the Embulk program?
Is it possible to fix the MySQL output plugin to use an INSERT that always does the right thing?

**Comment by frsyuki** [comment]
The right behavior could be either of:

* A) use a filter plugin to intentionally fill a non-NULL value (0) to "id" so that default behavior can safely make the transaction failed before loading broken records unintentionally. This is more conservative.
* B) change the default behavior of JDBC/MySQL output plugins so that they insert DEFAULT value (that is NULL in most of cases but non-NULL in some cases) if the value is NULL. This is more convenient.

Maybe B) is better. How do you think, @muga? 
In PostgreSQL, we can use `insert into t (id) values (DEFAULT)` syntax to insert a default value (that inserts NULL if DEFAULT option is not set to the column). I don't know how to do the same thing in MySQL.
